nbgrader/server_extensions/formgrader/handlers.py

```diff
@@ -3,164 +3,69 @@
+    @check_xsrf
     def get(self):
-        assignments = []
-        for assignment in self.gradebook.assignments:
-            x = assignment.to_dict()
-            x["average_score"] = self.gradebook.average_assignment_score(assignment.name)
-            x["average_code_score"] = self.gradebook.average_assignment_code_score(assignment.name)
-            x["average_written_score"] = self.gradebook.average_assignment_written_score(assignment.name)
-            assignments.append(x)
-
         html = self.render(
-            "assignments.tpl",
-            assignments=assignments,
+            "manage_assignments.tpl",
             base_url=self.base_url)
-
         self.write(html)
 
 
-class AssignmentNotebooksHandler(BaseHandler):
+class ManageSubmissionsHandler(BaseHandler):
     @web.authenticated
+    @check_xsrf
     def get(self, assignment_id):
-        try:
-            assignment = self.gradebook.find_assignment(assignment_id)
-        except MissingEntry:
-            raise web.HTTPError(404, "Invalid assignment: {}".format(assignment_id))
-
-        notebooks = []
-        for notebook in assignment.notebooks:
-            x = notebook.to_dict()
-            x["average_score"] = self.gradebook.average_notebook_score(notebook.name, assignment.name)
-            x["average_code_score"] = self.gradebook.average_notebook_code_score(notebook.name, assignment.name)
-            x["average_written_score"] = self.gradebook.average_notebook_written_score(notebook.name, assignment.name)
-            notebooks.append(x)
-        assignment = assignment.to_dict()
-
         html = self.render(
-            "assignment_notebooks.tpl",
-            assignment=assignment,
-            notebooks=notebooks,
-            base_url=self.base_url)
-
-        self.write(html)
-
-
-class AssignmentNotebookSubmissionsHandler(BaseHandler):
-    @web.authenticated
-    def get(self, assignment_id, notebook_id):
-        try:
-            self.gradebook.find_notebook(notebook_id, assignment_id)
-        except MissingEntry:
-            raise web.HTTPError(404, "Invalid notebook: {}/{}".format(assignment_id, notebook_id))
-
-        submissions = self.gradebook.notebook_submission_dicts(notebook_id, assignment_id)
-        indexes = self._notebook_submission_indexes(assignment_id, notebook_id)
-        for nb in submissions:
-            nb['index'] = indexes.get(nb['id'], None)
-
-        submissions = [x for x in submissions if x['index'] is not None]
-        submissions.sort(key=lambda x: x["id"])
-
-        html = self.render(
-            "notebook_submissions.tpl",
-            notebook_id=notebook_id,
+            "manage_submissions.tpl",
+            course_dir=self.coursedir.root,
             assignment_id=assignment_id,
-            submissions=submissions,
-            base_url=self.base_url
-        )
+            base_url=self.base_url)
         self.write(html)
 
 
-class StudentsHandler(BaseHandler):
+class Gradebook1Handler(BaseHandler):
```

Comment:
Question on the naming: why Gradebook1, 2, 3? Any way to better name what the Gradebook Handlers are doing?